packages/typespec-ts/src/modular/buildProjectFiles.ts

```diff
@@ -208,8 +208,8 @@ export function emitPackage(
     packageInfo.devDependencies["@azure-tools/test-credential"] = "^1.0.0";
     packageInfo.devDependencies["@azure/identity"] = "^2.0.1";
     packageInfo.devDependencies["@azure-tools/test-recorder"] = "^3.0.0";
-    packageInfo.devDependencies["mocha"] = "^7.1.1";
-    packageInfo.devDependencies["@types/mocha"] = "^7.0.2";
+    packageInfo.devDependencies["mocha"] = "^10.0.0";
```

Comment:
I'd like to confirm one thing which may be not relevant to this pr but impact the sample test file generations.

In current dev-tool we check there exists `type: module` in package.json to detect if this is a esm or cjs module, then if it is esm we should all follow the suffix `.js` within imports.

```
import { EventGridManagementClient } from "../src/eventGridManagementClient.js";
```

Reply:
> I'd like to confirm one thing which may be not relevant to this pr but impact the sample test file generations.

It would impact source code and test code, if we want to do it correctly in codegen when generate packages with `"type": "module"`.  We should generate relative imports with ".js" extension.

> dev-tool will not do any workaround to mitigate import errors if no js suffix. Is it correct or not?

we can add functionality to dev-tool to help add the .js extension for existing packages (generated or not).  I still think it's better to handled it in the codegen

Reply:
> we can add functionality to dev-tool to help add the .js extension for existing packages (generated or not). I still think it's better to handled it in the codegen

Yeah sure I also think we should handle it in codegen.

Reply:
yes, tracked by #1955